### PR TITLE
chore(actions): added automatic PR labels based on modified files

### LINKED ISSUE #62: Automatically apply technology/sgx label
Get a list of the files modified. Do a regular expression search for SGX. If found, label it with `technology/sgx`.

### PR DESCRIPTION
**Pull request by shravi24**
Closes #62

### CODE REVIEW

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+on:
+- pull_request_target
+
+jobs:
+  triage:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/labeler@v3
+      with:
+        repo-token: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (rvolosatovs) [inline .github/workflows/labeler.yml:11]: I guess this is "root token"? Not sure if there's anything better we can use (i.e. a scoped token of some kind)

**Review by rvolosatovs — APPROVED** [review]

File: .github/workflows/labeler.yml (hunk 1 above)
Comment (shravi24) [reply .github/workflows/labeler.yml:11]: We can use an equivalent here by creating specific token based on permissions and adding it as a secret in the repo settings. But not really sure how to do that.

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+cli:
+- src/cli/**/*
+
+protobuf:
+- src/protobuf/**/*
+
+workldr:
+- src/workldr/**/*
+
+tests/bin:
```
Comment (platten) [inline .github/labeler.yml:31]: Please remove.

File: .github/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+github:
```
Comment (platten) [inline .github/labeler.yml:1]: github --> infrastructure

File: .github/labeler.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+technology/wasm:
+- internal/wasmldr/**/*
+
+src:
+- src/**/*
+
+backend:
+- src/backend/**/*
+
+cli:
```
Comment (platten) [inline .github/labeler.yml:22]: `cli` --> `client`

File: .github/labeler.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+technolgy/sgx:
+- internal/shim-sgx/**/*
+
+technology/wasm:
+- internal/wasmldr/**/*
+
+src:
+- src/**/*
+
+backend:
```
Comment (platten) [inline .github/labeler.yml:19]: Please remove.

File: .github/labeler.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+- src/backend/**/*
+
+cli:
+- src/cli/**/*
+
+protobuf:
+- src/protobuf/**/*
+
+workldr:
```
Comment (platten) [inline .github/labeler.yml:28]: Please remove.

File: .github/labeler.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+src:
+- src/**/*
+
+backend:
+- src/backend/**/*
+
+cli:
+- src/cli/**/*
+
+protobuf:
```
Comment (platten) [inline .github/labeler.yml:25]: Please remove.

File: .github/labeler.yml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+- src/protobuf/**/*
+
+workldr:
+- src/workldr/**/*
+
+tests/bin:
+- tests/bin/**/*
+
+tests/common:
```
Comment (platten) [inline .github/labeler.yml:34]: Please remove.

File: .github/labeler.yml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+- tests/common/**/*
+
+tests/wasm:
```
Comment (platten) [inline .github/labeler.yml:37]: Please have this labeled with both `wasm` as well as `testing`

File: .github/labeler.yml [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+protobuf:
+- src/protobuf/**/*
+
+workldr:
+- src/workldr/**/*
+
+tests/bin:
+- tests/bin/**/*
+
```
Comment (platten) [inline .github/labeler.yml:33]: I would recommend simply labeling all of the content in `tests/**/*`  as `testing`.

**Review by platten — CHANGES_REQUESTED** [review]
Hello @shravi24 just a few changes, thanks!

File: .github/labeler.yml (hunk 10 above)
Comment (shravi24) [reply .github/labeler.yml:33]: Sure thing!!

**Review by platten — APPROVED** [review]
LGTM

**Review by platten — CHANGES_REQUESTED** [review]
Hello @shravi24, could you please rebase, squash and update your commit messages & PR title and to be compliant with [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).

File: .github/labeler.yml [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+wasm:
+- internal/wasmldr/**/*
+- tests/wasm/**/*
+
+client:
+- src/cli/**/*
+
+testing:
+- tests/**/*
```
Comment (platten) [inline .github/labeler.yml:19]: Please add an empty newline at the end of the file.

File: .github/workflows/labeler.yml [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+on:
+- pull_request_target
+
+jobs:
+  triage:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/labeler@v4
+      with:
+        repo-token: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (platten) [inline .github/workflows/labeler.yml:11]: Please add an empty newline at the end of the file.

**Review by platten — CHANGES_REQUESTED** [review]
Please add those newlines at. the end of the files, rebase, squash the commits, and you should be good! Thank you!

**Review by platten — CHANGES_REQUESTED** [review]
Please squash the commits into a single commit with a [conventional commit compliant message](https://enarx.dev/docs/Contributing/PRs#commit-messages), and [rebase](https://enarx.dev/docs/Contributing/PRs#rebase-onto-upstreammain). On the enarx project we require a linear history and thus mandate the use of rebase. Thank you again for pushing this over the line!

**Comment by platten** [comment]
@shravi24 I took the liberty of recreating and rebasing your commits in PR #1654

**Pull request closed without merging**